sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
Your code compiles fine for me. I've gone through it step by step and it works as supposed.

I've managed to reproduce and fix the bug you mention. Is happens when the additional files weren't read and is easily fixed by initializing **msg** variable with nullptr/NULL when it is defined in **op_message_str_game2** function. By the way, I found an unnecessary line of code and commented on it below.

Looks like your files weren't properly read. Are you using an English version of the game? I think that engine file reading functions will look up different directories based on the language of the game.